platform/android/src/string_util.cpp

```diff
@@ -0,0 +1,27 @@
+#include <mbgl/util/platform.hpp>
+#include "attach_env.hpp"
+#include <jni/jni.hpp>
+
+namespace mbgl {
+namespace platform {
+
+std::string uppercase(const std::string& str) {
+    auto env{ android::AttachEnv() };
+    jni::Local<jni::String> value = jni::Make<jni::String>(*env, str.c_str());
+    static auto& javaClass = jni::Class<jni::StringTag>::Singleton(*env);
```

Comment:
Let's fold this line into the next to avoid the need for another static variable. Same for `lowercase`